src/core_plugins/elasticsearch/lib/clusters.js

```diff
@@ -0,0 +1,14 @@
+import Cluster from './cluster';
+
+const clusters = new Map();
```

Comment:
I'm more worried about the cluster objects not having the same lifecycle as the server. If the server is closed and reopened, what happens here? What happens if multiple servers run in the same thread and both register `'data'` clusters.